pages/api/token/temporary.ts

```diff
@@ -143,4 +160,114 @@ const sendTemporaryTokenByEmail = async (email: string, temporaryToken: string)
     });
 };
 
-export default middleware([cors({ allowedMethods: ["POST"] }), validBearerToken()], handler);
+const requestContainsEmail = (): MiddlewareRequest => {
+  return async (req: NextApiRequest, res: NextApiResponse): Promise<MiddlewareReturn> => {
+    const email = req.body["email"];
+
+    if (!email) {
+      res.status(400).json({ error: "Invalid payload" });
+      return { next: false };
+    }
+
+    return { next: true, props: { email } };
+  };
+};
+
+const canLogIn = (): MiddlewareRequest => {
+  return async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+    props?: MiddlewareProps
+  ): Promise<MiddlewareReturn> => {
+    assert(props?.email); // Middleware function `requestContainsEmail` ensures that email is defined
+
+    const loginLockoutResponse = await isUserAllowedToLogIn(props.email);
+
+    if (loginLockoutResponse.isAllowedToLogIn) return { next: true };
+    else {
+      res
+        .status(401)
+        .json(
+          buildUserNotAllowedToLogInErrorMessage(
+            loginLockoutResponse.remainingNumberOfAttemptsBeforeLockout,
+            loginLockoutResponse.numberOfSecondsBeforeLockoutExpires
+          )
+        );
+      return { next: false };
+    }
+  };
+};
+
+/**
+ * This is a middleware function that will validate the bearer token in the authorization header
+ *
+ * @param handler - the function to be executed next in the API call
+ * @returns either the handler to be executed next in the API call, or updates the res status and returns void
+ */
+export const hasValidFormAccessToken = (): MiddlewareRequest => {
```

Comment:
As with the other middleware comments this should probably be moved back to `validBearerToken` middleware.